templates/todo/api/csharp-cosmos-sql/TodoEndPointsExt.cs

```diff
@@ -0,0 +1,143 @@
+            {
+                var existingList = await _repository.GetListAsync(list_id);
+                if (existingList == null)
+                {
+                    return TypedResults.NotFound();
+                }
+
+                existingList.Name = list.name;
+                existingList.Description = list.description;
+                existingList.UpdatedDate = DateTimeOffset.UtcNow;
+
+                await _repository.UpdateList(existingList);
+
+                return TypedResults.Ok(existingList);
+            }).Produces(200).WithOpenApi();
+
+            app.MapDelete("/lists/{list_id}", async (ListsRepository _repository, string list_id) => {
+                if (await _repository.GetListAsync(list_id) == null)
+                {
+                    return Results.NotFound();
+                }
+
+                await _repository.DeleteListAsync(list_id);
+
+                return Results.NoContent();
+            }).Produces(204).WithOpenApi();
+
+
+            app.MapGet("/lists/{list_id}/items", async (ListsRepository _repository, string list_id, [FromQuery] int? skip = null, [FromQuery] int? batchSize = null) =>
+            {
+                return TypedResults.Ok(await _repository.GetListItemsAsync(list_id, skip, batchSize));
+            }).Produces(200).WithOpenApi();
+
+
+            app.MapPost("/lists/{list_id}/items", async (ListsRepository _repository, string list_id, [FromBody] CreateUpdateTodoItem item) =>
+            {
+                var newItem = new TodoItem(list_id, item.name)
+                {
+                    Name = item.name,
+                    Description = item.description,
+                    State = item.state,
+                    CreatedDate = DateTimeOffset.UtcNow
+                };
+
+                await _repository.AddListItemAsync(newItem);
+                return Results.CreatedAtRoute($"/lists/{list_id}/items", new { list_id = list_id, item_id = newItem.Id }, newItem);
+
+            }).Produces(201).WithOpenApi();
+
+            app.MapGet("/lists/{list_id}/items/{item_id}", async Task<Results<Ok<TodoItem>, NotFound>> (ListsRepository _repository, string list_id, string item_id) =>
+            {
+                var item = await _repository.GetListItemAsync(list_id, item_id);
+
+                return item == null ? TypedResults.NotFound() : TypedResults.Ok(item);
+            }).Produces(200).WithOpenApi();
+
+
+            app.MapPut("/lists/{list_id}/items/{item_id}", async Task<Results<Ok<TodoItem>, NotFound>> (ListsRepository _repository, string list_id, string item_id, [FromBody] CreateUpdateTodoItem item) =>
+            {
+                var existingItem = await _repository.GetListItemAsync(list_id, item_id);
+                if (existingItem == null)
+                {
+                    return TypedResults.NotFound();
+                }
+
+                existingItem.Name = item.name;

```

Comment:
Perhaps we should implement a copy constructor on `CreateUpdateTodoItem` and make it a class?

Reply:
Not sure how that would work as you eventually need to convert back to TodoItem for the datasource. Unless I am misunderstanding what you are suggesting

Reply:
resolving per our chat on Teams